router/internal/graphiql/playgroundhandler.go

```diff
@@ -35,12 +44,9 @@ func (p *Playground) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	tpl := strings.Replace(p.opts.Html, "{{graphqlURL}}", p.opts.GraphqlURL, -1)
-	resp := []byte(tpl)
-
 	w.Header().Set("Content-Type", "text/html; charset=utf-8")
-	w.Header().Set("Content-Length", strconv.Itoa(len(resp)))
+	w.Header().Set("Content-Length", strconv.Itoa(len(*p.templateBytes)))
 
 	w.WriteHeader(http.StatusOK)
-	_, _ = w.Write(resp)
+	_, _ = w.Write(*p.templateBytes)
```

Comment:
I think I'd do this instead:
```go
_, _ = bytes.NewBuffer(resp).WriteTo(w)
```

Explanation:

Internally, the net/http package has a 4kb buffer. The response size of the HTML is 3mb. As such, the internal buffer will have to expand (grow -> copy) multiple times to reach 3mb. Initializing the buffer with 3mb will eliminate the necessary grow calls.

In addition, this could become a security concern as an attacker could overwhelm the router just by sending a lot of traffic to this endpoint, as it will create a lot of buffers.

To mitigate this, we could add a simple semaphore e.g. to limit the concurrency in this handler to 10, which will limit the memory usage to 30mb.